asdf/commands/edit.py

```diff
@@ -0,0 +1,529 @@
+"""
+
+import io
+import os
+import struct
+import sys
+
+import asdf.constants as constants
+
+from asdf.asdf import _parse_asdf_header_line
+from asdf.asdf import _parse_asdf_comment_section
+from asdf.asdf import _get_asdf_version_in_comments
+
+from .. import AsdfFile
+from .. import generic_io
+from .. import reference
+from .. import schema
+from .. import yamlutil
+
+from .main import Command
+
+__all__ = ['edit']
+
+#asdf_format_version = None
+#asdf_standard_version = None
+
+
+class Edit(Command):
+    @classmethod
+    def setup_arguments(cls, subparsers):
+        """ Set up a command line argument parser for the edit subcommand.
+        """
+        desc_string = "Allows for easy editing of the YAML in an ASDF file.  " \
+                      "For edit mode, the YAML portion of an ASDF file is"  \
+                      "separated from the ASDF into a text file for easy" \
+                      "editing.  For save mode, the edited text file is written" \
+                      "to its ASDF file."
+
+        # Set up the parser
+        parser = subparsers.add_parser(
+            str("edit"), help="Edit YAML portion of an ASDF file.",
+            description=desc_string)
+
+        # Need an input file
+        parser.add_argument(
+            '--infile', '-f', type=str, required=True, dest='fname',
+            help="Input file (ASDF for -e option, YAML for -s option")
+
+        # Need an output file
+        parser.add_argument(
+            '--outfile', '-o', type=str, required=True, dest='oname',
+            help="Output file (YAML for -e option, ASDF for -s option")
+
+        # The edit is either being performed or saved
+        group = parser.add_mutually_exclusive_group(required=True)
+        group.add_argument(
+            '-s',action='store_true',dest='save',
+            help="Saves a YAML text file to its ASDF file.  Requires an ASDF input file.")
+        group.add_argument(
+            '-e',action='store_true',dest='edit',
+            help="Create a YAML text file for a ASDF file.  Requires a YAML input file.")
+
+        parser.set_defaults(func=cls.run)
+
+        return parser
+
+    @classmethod
+    def run(cls, args):
+        """ Execute the edit subcommand.
+        """
+        return edit(args)
+
+
+def is_yaml_file ( fname ) :
+    '''
+    Determines if a file is a YAML file based only on the file extension.
+
+    Parameters
+    ----------
+    fname : The input file name.
+    '''
+
+    base, ext = os.path.splitext(fname)
+    if '.yaml' != ext :
+        return False
+    return True
+
+
+def is_asdf_file ( fname ) :
+    '''
+    Determines if a file is ASDF based on file extension and the first
+    5 bytes of the file, which should be '#ASDF'.
+
+    Parameters
+    ----------
+    fname : The input file name.
+    '''
+
+    base, ext = os.path.splitext(fname)
+    if '.asdf' != ext :
+        return False
+
+    with open(fname,"r+b") as fd :
+        first_string = "#ASDF"
```

Comment:
I may be a bit too picky on this, but I see a number of cases where there is variable defined from a constant, then that constant is used in a test. Unless that constant is subject to change (in which case it belongs with other such constants), it would just be simpler to put that string in the test. At least in my opinion. Others may not agree with me :-)

Reply:
I mention this down below, but in this case there is a constant defined in constants.py that we can use.

Reply:
I changed this to use the string defined in constants.